addons/edi_notifier/models/notifier.py

```diff
@@ -168,12 +169,28 @@ def action_view_cron(self):
         return action
 
     @api.multi
-    def notify(self, event_type, recs=None):
-        for notifier in self:
-            if notifier.active:
-                self.env[notifier.model_id.model].notify(
-                    notifier, event_type, recs
+    def check_edi_notifications_enabled(self):
+        """Check safety config parameter is present and return True if it's enable, otherwise False."""
+        self.ensure_one()
+        if self.safety:
+            section, _sep, key = self.safety.rpartition(".")
+            if not config.get_misc(section or self._name, key):
+                _logger.info(
+                    "%s %s disabled, enable by configuring safety %s",
+                    self._name,
+                    self.name,
+                    ".".join((section or self._name, key)),
                 )
+                return False
+        return True
+
+    @api.multi
+    def notify(self, event_type, recs=None):
+        """Check for edi notification safety"""
+        if self.check_edi_notifications_enabled():
```

Comment:
his check should be inside the loop, otherwise this function allows self to be nothing or more than one element in the recordset, and `check_edi_notifications_enabled()` ensures one.